tfx/utils/io_utils.py

```diff
@@ -77,15 +78,26 @@ def copy_dir(src: Text, dst: Text) -> None:
       fileio.makedirs(os.path.join(dir_name.replace(src, dst, 1), sub_dir))
 
 
-def get_only_uri_in_dir(dir_path: Text) -> Text:
-  """Gets the only uri from given directory."""
+def get_only_uri_in_dir(dir_path: Text, uri_pattern: Text = None) -> Text:
```

Comment:
It might be worth renaming this one, since even those it is compatible with the old implementation when no file pattern is supplied, the semantics are different when there is one. Maybe `get_only_uri_matching_pattern`? And maybe the error message below can be adjusted too.

Reply:
Yeah, so, with this, I hear you, but I think I'm going to leave it as-is for now. When I implemented it this way, my logic was the following. I use this function, and I know that the structure of my artifact is always expected to be `/some_dir/single_uri_node,` then this name works. Or, let's say, I used this function, but it didn't work for me because there is some unexpected metadata under `/some_dir,` but I still know that there is a single target artifact node under that dir. A natural thing for me to do is find a better function for my case or check whether `get_only_uri_in_dir` has any useful params. Instead of implementing another function with a more descriptive name, I decided to add a param because it seems that the name still fundamentally works. It conveys the expectation of having a single artifact node in a way. Does it make sense?

And another reason, I'm a little wary of renaming functions in general. In this case, because IMO the name still works (naming is hard!), I think I'll keep it adjusted at the signature level unless they suggest doing so.

That said, how strongly do you feel about renaming? cc: @sngahane

Reply:
Not strong, SGTM

Reply:
Maybe rephrasing the error message will still be good though, but I'll wait for @andrewsmartin to respond first.

Reply:
I think your explanation makes sense @yanyuchka, that given a pattern name as an argument, it should still capture that only a single file should be matched. And the point about compatibility is a good one, there are probably several people out there  writing custom components that make use of this function, so the tradeoff of a slightly-clearer-name probably isn't worth breaking compatibility. +1 to keeping as is!